satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -459,6 +457,29 @@ def __init__(self, nc):
         """Wrap the given dataset."""
         self.nc = nc
 
+        # remove time before decoding and add again.
+        raw_time = nc["time_ir_wv"]
+        self.nc = self.nc.drop_vars(["time_ir_wv"])
+        self.nc = xr.decode_cf(self.nc)
+        self.nc["time_ir_wv"] = raw_time
```

Comment:
Can you please extract that into a separate method, for example `DatasetWrapper._decode_cf()`

Reply:
BTW, I just noticed that in `get_time` there's a note:

> Variable is sometimes named "time" and sometimes "time_ir_wv"

I guess this would have to be handled here, too.

Reply:
Maybe just renaming to a common time variable name is sufficient